web/src/composebox_typeahead.ts

```diff
@@ -1008,8 +1034,25 @@ export function content_highlighter_html(item: TypeaheadSuggestion): string | un
             return typeahead_helper.render_typeahead_item({primary: item.language});
         case "topic_jump":
             return typeahead_helper.render_typeahead_item({primary: item.message});
-        case "topic_list":
-            return typeahead_helper.render_typeahead_item({primary: item.topic});
+        case "topic_list": {
+            if (item.is_stream && item.streamData) {
+                item.streamData.rendered_description = "";
+                return typeahead_helper.render_stream(item.streamData);
+            }
+
+            let stream_html = typeahead_helper.render_stream(current_stream_data!);
+            stream_html = stream_html.replace(
+                /<div class="typeahead-text-container has_secondary_html"/,
+                '<div class="typeahead-text-container has_secondary_html compose-stream-name" ',
+            );
+            return (
+                stream_html +
+                `
+            <span role="button" class="conversation-arrow zulip-icon zulip-icon-chevron-right"
+            ></span>` +
+                typeahead_helper.render_typeahead_item({primary: item.topic})
+            );
```

Comment:
Don't edit html in javascript files. You probably want to update `typeahead_list_item` to have  a separate portion for rendering the stream>topic, that gets called similarly to all the other other list items. This might be a bit complicated, so let me know if you want help getting this set up.